pkg/kubelet/metrics/metrics.go

```diff
@@ -275,13 +295,17 @@ func (pc *podAndContainerCollector) Collect(ch chan<- prometheus.Metric) {
 	}
 
 	runningContainers := 0
+	podsByRuntimeClassName := make(map[string]float64)
 	for _, p := range runningPods {
 		runningContainers += len(p.Containers)
+		podsByRuntimeClassName[p.RuntimeClassName] += 1
```

Comment:
Oops, `p.RuntimeClassName` will not work because [Pod](https://github.com/kubernetes/kubernetes/blob/0f93328c7a051e28a097270daaf7a7ff6f90bae0/pkg/kubelet/container/runtime.go#L163) does not have a `RuntimeClassName` field.

Reply:
Hmm, ick. The easiest way to fix this would be to add the RuntimeClass to the pod annotations, but I'm not sure what the implications of that are - @yujuhong would appreciate your thoughts on this.

The other alternative is to maintain a cache with a pod UID -> runtimeclass mapping in the RuntimeClassManager, but we need to consider what happens to that if the kubelet restarts.

Reply:
Either of these options become a bigger change, so it's probably worth separating out this metric into a separate PR.

Reply:
I separated this metric out for now.